src/lib/hooks/resource-names.svelte.ts

```diff
@@ -72,15 +72,17 @@ const sortResourceNames = (resourceNames: ResourceName[]) => {
 };
 
 export const provideResourceNamesContext = () => {
+  const connectionStatuses = useConnectionStatuses();
   const machineStatuses = useMachineStatuses();
   const clients = useRobotClients();
 
   const partIDs = $derived(Object.keys(clients.current));
-  const options = $derived(
-    partIDs.map((partID) => {
```

Comment:
Another reactivity fix, since the map function is a different closure `client` below is not reactive